src/android/android_interface.rs

```diff
@@ -27,12 +41,258 @@ pub unsafe extern "C" fn Java_org_coepi_android_api_NativeApi_bootstrapCore(
     let db_path_java_str = env.get_string(db_path_j_string).unwrap();
     let db_path_str = db_path_java_str.to_str().map_err(ServicesError::from);
     info!("Bootstrapping with db path: {:?}", db_path_str);
-    let db_result = db_path_str.and_then(|path| init_db(path).map_err(ServicesError::from));
+    let db_result = db_path_str.and_then(|path| bootstrap(path).map_err(ServicesError::from));
     info!("Bootstrapping result: {:?}", db_result);
 
     jni_void_result(1, None, &env)
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn Java_org_coepi_android_api_NativeApi_fetchNewReports(
+    env: JNIEnv,
+    _: JClass,
+) -> jobject {
+    info!("Updating reports");
+    // TODO error handling https://github.com/Co-Epi/app-backend-rust/issues/79
+    let result = dependencies().reports_updater.fetch_new_reports().unwrap();
+    info!("New reports: {:?}", result);
+
+    let alerts_j_objects: Vec<jobject> = result
+        .into_iter()
+        .map(|alert| alert_to_jobject(alert, &env))
+        .collect();
+
+    let placeholder_alert_j_object = alert_to_jobject(placeholder_alert(), &env);
+
+    let alerts_array = env
+        .new_object_array(
+            alerts_j_objects.len() as i32,
+            "org/coepi/android/api/JniAlert",
+            placeholder_alert_j_object,
+        )
+        .unwrap();
+
+    for (index, alert_j_object) in alerts_j_objects.into_iter().enumerate() {
+        env.set_object_array_element(alerts_array, index as i32, alert_j_object)
+            .unwrap();
+    }
+
+    jni_obj_result(
+        1,
+        None,
+        JObject::from(alerts_array),
+        "org/coepi/android/api/JniAlertsArrayResult",
+        "[Lorg/coepi/android/api/JniAlert;",
+        &env,
+    )
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn Java_org_coepi_android_api_NativeApi_recordTcn(
+    env: JNIEnv,
+    _: JClass,
+    tcn: JString,
+) -> jobject {
+    let tcn_java_str = env.get_string(tcn).unwrap();
```

Comment:
Should we avoid unwrap?

Reply:
Like last time, The error handling improvements will be in a separate PR (see [issue](https://github.com/Co-Epi/app-backend-rust/issues/79). It should only affect this file. I want to finish the app integration ASAP.